pkg/core/resources/apis/meshservice/api/v1alpha1/meshservice.go

```diff
@@ -0,0 +1,57 @@
+// +kubebuilder:object:generate=true
+package v1alpha1
+
+import (
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+)
+
+type DataplaneTags map[string]string
+
+type Selector struct {
+	DataplaneTags DataplaneTags `json:"dataplaneTags,omitempty"`
+}
+
+type Port struct {
+	Port       uint32 `json:"port"`
+	TargetPort uint32 `json:"targetPort,omitempty"`
+	// +kubebuilder:default=tcp
+	Protocol core_mesh.Protocol `json:"protocol,omitempty"`
+}
+
+// MeshService
+// +kuma:policy:is_policy=false
+type MeshService struct {
+	Selector Selector `json:"selector,omitempty"`
+	// +patchMergeKey=port
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=port
+	// +listMapKey=protocol
+	Ports []Port `json:"ports,omitempty"`
+
+	Status MeshServiceStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=Available;NotAvailable
+type AddressStatus string
```

Comment:
should you already create the entries for the enums?

Reply:
removed this status for now